lcd/lcd_api.py

```diff
@@ -140,7 +140,8 @@ def putchar(self, char):
             self.cursor_x += 1
         if self.cursor_x >= self.num_columns or char == '\n':
             self.cursor_x = 0
-            self.cursor_y += 1
+            if char == '\n':
+                self.cursor_y += 1
```

Comment:
It seems to me that if you output to exactly fill the display without a \n then this change will leave the cursor at the beginning of the last line rather than at the beginning of line 0.

I'll have to write some test code to see what happens.

Reply:
With your proposed change, the following test:
```
    lcd.putstr('Line 1 - 01234567890')
    lcd.putstr('Line 2 - 01234567890')
    lcd.putstr('Line 3 - 01234567890')
    lcd.putstr('Line 4 - 01234567890')
    lcd.putstr('Line 5 - 01234567890')
```
results in all 5 lines printing on top of each other on the first line. Without this change, Line 5 overwrites Line 1 (which is what I would have expected).

Can you provide an example test and explain what you're seeing with and without the patch?

Reply:
I see the problem: If I use this:
```
    lcd.putstr('Line 1 - 01234567890\n')
    lcd.putstr('Line 2 - 01234567890\n')
    lcd.putstr('Line 3 - 01234567890\n')
    lcd.putstr('Line 4 - 01234567890\n')
    lcd.putstr('Line 5 - 01234567890\n')
```
then I get the blank lines like you describe.

Reply:
I fixed the issue via 18bc2e07bbdb5d0748da6a150f8a4f305d77405d